### PR TITLE
Nikolai/feature/tag class

### PR DESCRIPTION
**Pull request by lizard-52**
Add some classes to deal with tags. I also made Tags store a Color object, but some colour methods require API 26 (current min is 24). Do we want to change the minimum up to 26?

### CODE REVIEW

File: code/app/src/main/java/com/example/sigma_blue/TagDB.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Ideally this class will only be used from TagList.
+ */
+public class TagDB {
+    /**
+     * Update the database with the current tags
+     */
+    public void saveToDB(TagList tList) {};
+
+    /**
+     * Update the
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/TagDB.java:17]: I think that this documentation might not be done.

File: code/app/src/main/java/com/example/sigma_blue/Tag.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    public int compareTo(Tag o) {
+        return this.tagText.compareTo(o.getTagText());
+    }
+
+    @Override
+    public int hashCode() {
+        return this.tagText.hashCode();
+    }
+
+    @Override
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/Tag.java:44]: Should we do a color comparison as well? Should that be left for future implementation?

File: code/app/src/main/java/com/example/sigma_blue/Tag.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+import android.graphics.Color;
+
+/**
+ * Stores information about a single tag.
+ */
+public class Tag implements Comparable<Tag>{
+    private String tagText;
+    private Color colour;
+
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/Tag.java:11]: Should we have a colorless implementation where the Tag is constructed without a Color?

File: code/app/src/main/java/com/example/sigma_blue/Tag.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public void setTagText(String tagText) {
+        this.tagText = tagText;
+    }
+
+    public Color getColour() {
+        return colour;
+    }
+
+    @Override
+    public int compareTo(Tag o) {
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/Tag.java:35]: Should we have a color comparison if the text returns equals (0)?

File: code/app/src/main/java/com/example/sigma_blue/TagList.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    public void refreshTagsFromDB() {
+        this.tags = tagDB.refreshFromDB();
+    }
+
+    public void addTag(Tag tag) {
+        tags.add(tag);
+    }
+
+    public boolean containsTag(Tag tag) {
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/TagList.java:28]: How is our handling of null input? Is that an exception thing? Are we trying to not throw any nulls at all? (I think eventually, nulls will get thrown, and we should do something about it.)

**Review by waridh — APPROVED** [review]
I left comments about what we could think about or potentially work on.

Approving for now since this will not break anything.

File: code/app/src/main/java/com/example/sigma_blue/Tag.java (hunk 2 above)
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/Tag.java:44]: I think having two tags with the same name but different colours is a bad use case.

File: code/app/src/main/java/com/example/sigma_blue/Tag.java (hunk 3 above)
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/Tag.java:11]: I mean we could set it to white by default

File: code/app/src/main/java/com/example/sigma_blue/TagList.java (hunk 5 above)
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/TagList.java:28]: Not sure. maybe we should talk about this later today

File: code/app/src/main/java/com/example/sigma_blue/Tag.java (hunk 3 above)
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/Tag.java:11]: I think a better solution is to just use the constructor with an int and use new Tag("name", WHITE). I don't think it really justifies another constructor.

Comment (waridh) [reply code/app/src/main/java/com/example/sigma_blue/Tag.java:13]: > I think a better solution is to just use the constructor with an int and use new Tag("name", WHITE). I don't think it really justifies another constructor.

Might be something that the factory pattern could help with, but might not be necessary as well.

**Review by waridh — APPROVED** [review]
LGTM

**Review by Yunyang-Lu — APPROVED** [review]

**Pull request merged**